be/src/exprs/cast_expr.cpp

```diff
@@ -1621,6 +1621,9 @@ StatusOr<std::unique_ptr<Expr>> VectorizedCastExprFactory::create_cast_expr(Obje
         if (from_type.children.size() != to_type.children.size()) {
             return Status::NotSupported("vectorized engine not support cast struct with different number of children.");
         }
+        if (to_type.field_names.empty() || from_type.field_names.size() != to_type.field_names.size()) {
```

Comment:
FE has similar check. So is it for stablity?

Reply:
check strict, avoid crash in upgrade